tiledb/cloud/dag/visualization.py

```diff
@@ -0,0 +1,230 @@
+import networkx as nx
+import random
+
+from . import dag
+
+
+def build_graph_node_details(nodes):
+    """
+    :param nodes: List of nodes to get status of
+    :return: tuple of node_colors and node_text
+    """
+    # Loop over statuses to set color and label.
+    # If you rerun this cell and the one below you can update the graph
+    node_colors = []
+    node_text = []
+    for node in nodes:
+        status = node.status
+        if status == dag.Status.NOT_STARTED:
+            node_text.append("{} - Not Started".format(node.name))
+            node_colors.append("black")
+        elif status == dag.Status.RUNNING:
+            node_text.append("{} - Running".format(node.name))
+            node_colors.append("blue")
+        elif status == dag.Status.COMPLETED:
+            node_text.append("{} - Completed".format(node.name))
+            node_colors.append("green")
+        elif status == dag.Status.FAILED:
+            node_text.append("{} - Failed".format(node.name))
+            node_colors.append("red")
+        elif status == dag.Status.CANCELLED:
+            node_text.append("{} - Cancelled".format(node.name))
+            node_colors.append("yellow")
+
+    return (node_colors, node_text)
+
+
+def update_graph(nodes, fig=None):
+    """
+    Update a graph based on based node status and figure
+    :param nodes: list of notes to update
+    :param fig:
+    :return:
+    """
+
+    (node_colors, node_text) = build_graph_node_details(nodes)
+
+    if fig is not None:
+        fig.update_traces(
+            marker=dict(color=node_colors),
+            text=node_text,
+            selector=dict(mode="markers"),
+        )
+
+
+def hierarchy_pos(
+    G, root=None, width=1.0, vert_gap=0.2, vert_loc=0, leaf_vs_root_factor=0.5
+):
+    """
+    Taken from https://epidemicsonnetworks.readthedocs.io/en/latest/_modules/EoN/auxiliary.html#hierarchy_pos
```

Comment:
Should explicitly mention that it is MIT licensed for future clarity: https://epidemicsonnetworks.readthedocs.io/en/latest/_downloads/8e9c8138fef49ddba8102fa7799c29d7/license.txt

Reply:
Done, thanks for point it out.